argilla/src/argilla/datasets/_export/_hub.py

```diff
@@ -0,0 +1,199 @@
+                    f" are: {', '.join(hf_dataset.keys())}."
+                )
+            hf_dataset: Dataset = hf_dataset[list(hf_dataset.keys())[0]]
+
+        if with_records:
+            cls._log_dataset_records(hf_dataset=hf_dataset, dataset=dataset)
+        return dataset
+
+    @staticmethod
+    def _log_dataset_records(hf_dataset: "HFDataset", dataset: "Dataset"):
+        """This method extracts the responses from a Hugging Face dataset and returns a list of `Record` objects"""
+
+        # Identify columns that colunms that contain responses
+        responses_columns = [col for col in hf_dataset.column_names if ".responses" in col]
+        response_questions = defaultdict(dict)
+        user_ids = {}
+        for col in responses_columns:
+            question_name = col.split(".")[0]
+            if col.endswith("users"):
+                response_questions[question_name]["users"] = hf_dataset[col]
+                user_ids.update({UUID(user_id): UUID(user_id) for user_id in set(sum(hf_dataset[col], []))})
+            elif col.endswith("responses"):
+                response_questions[question_name]["responses"] = hf_dataset[col]
+            elif col.endswith("status"):
+                response_questions[question_name]["status"] = hf_dataset[col]
+
+        # Check if all user ids are known to this Argilla client
+        known_users_ids = [user.id for user in dataset._client.users._api.list()]
+        unknown_user_ids = set(user_ids.keys()) - set(known_users_ids)
+        if len(unknown_user_ids) > 1:
+            warnings.warn(
+                message=f"""Found unknown user ids in dataset repo: {unknown_user_ids}.
+                    Assigning first response for each record to current user ({dataset._client.me.username}) and discarding the rest."""
+            )
+        for unknown_user_id in unknown_user_ids:
+            user_ids[unknown_user_id] = dataset._client.me.id
+
+        # Create a mapper to map the Hugging Face dataset to a Record object
+        mapping = {col: col for col in hf_dataset.column_names if ".suggestion" in col}
+        mapper = IngestedRecordMapper(dataset=dataset, mapping=mapping, user_id=dataset._client.me.id)
+
+        # Extract responses and create Record objects
+        records = []
+        for idx, row in enumerate(hf_dataset):
+            record = mapper(row)
+            record.id = row.pop("id")
+            for question_name, values in response_questions.items():
+                _user_ids = user_ids.copy()
+                response_values = values["responses"][idx]
+                response_users = values["users"][idx]
+                response_status = values["status"][idx]
+                for value, user_id, status in zip(response_values, response_users, response_status):
+                    user_id = _user_ids.pop(UUID(user_id), None)
+                    if user_id is None:
```

Comment:
I think this change allows what you want: If a user is already used (is in responses_users), the response is skipped.
```suggestion
        for idx, row in enumerate(hf_dataset):
            record = mapper(row)
            record.id = row.pop("id")
            for question_name, values in response_questions.items():
                response_users = {}
                response_values = values["responses"][idx]
                response_users = values["users"][idx]
                response_status = values["status"][idx]
                for value, user_id, status in zip(response_values, response_users, response_status):
                    user_id = user_ids[UUID(user_id)]
                    if user_id is in response_users:
                        continue
                    response_users[user_id] = True   
```